app/models/like.rb

```diff
@@ -0,0 +1,11 @@
+class Like < ApplicationRecord
+  attribute :author_id, :integer
+  attribute :post_id, :integer
+
+  belongs_to :post, counter_cache: true
+  belongs_to :author, class_name: 'User'
+
+  def update_post_likes_count
+    post.update(likes_count: post.likes.count)
+  end
+end
```

Comment:
- [ **OPTIONAL** ] After, making the changes I mentioned above, I think it would be better if you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_likes_count` method every time a new like has been saved to database, then make the `update_post_likes_count` method private. I can also suggest you apply this approach in both `Post` and `Comment` models on the `update_counts` mthods. Below, is an example.
```ruby
class Like < ApplicationRecord
 ...
 after_save :update_post_likes_count

  private

  def update_post_likes_count
    author.increment!(:posts_counter)
  end
end
```

Reply:
Thanks for your comments, however, I think it is a better option to use counter_cache, as you can see I am using it however I also created the methods required by the project requirement.
I don't think the option you suggest is to use the increment! is so appropriate since it doesn't take into account when an object is deleted. instead counter_cache handles everything automatically. without writing a lot of code.